pkg/machines/components/vmSnapshotsTab.jsx

```diff
@@ -0,0 +1,174 @@
+            sameElse : "L LT"
+        }
+    });
+
+    return moment(Number(unixTime) * 1000).calendar();
+}
+
+class VmSnapshotsTab extends React.Component {
+    render() {
+        const { vm } = this.props;
+        const id = vmId(vm.name);
+
+        const emptyCaption = (<div className="no-snapshots">
+            <h3>{_("No snapshots")}</h3>
+            {_("Snapshots allow you to revert to an earlier state if something goes wrong")}
+        </div>);
+
+        let detailMap = [
+            {
+                name: _("Time"), value: (snap, snapId) => {
+                    const date = prettyTime(snap.creationTime);
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-date`}>
+                            {date}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Name"), value: (snap, snapId) => {
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-name`}>
+                            {snap.name}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Description"), value: (snap, snapId) => {
+                    let desc = snap.description;
+                    if (!desc)
+                        desc = (<span className="snap-greyed-out">{_("No description")}</span>);
+
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-description`}>
+                            {desc}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("State"), value: (snap, snapId) => {
+                    const state = snap.state == "shutoff" ? "shut off" : snap.state;
+
+                    const infoTips = {
+                        shutdown: _("Shutting down"),
+                        "disk-snapshot": _("Disk-only snapshot"),
+                        blocked: _("Domain is blocked on resource"),
+                        crashed: _("Domain has crashed"),
+                    };
+                    const tooltipMessage = infoTips[state];
+                    const tooltip = tooltipMessage
+                        ? (<span className="tooltip-circle">
+                            <Tooltip entryDelay={0} exitDelay={0} content={tooltipMessage}>
+                                <InfoAltIcon />
+                            </Tooltip>
+                        </span>) : null;
+
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-type`}>
+                            {snap.state}
+                            {tooltip}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Parent snapshot"), value: (snap, snapId) => {
+                    let parentName = snap.parentName;
+                    if (!parentName)
```

Comment:
You can make it oneliner with:
const parentName = snap.parentName || <span className="snap-greyed-out">{_("No parent")}</span>;